packages/dds/tree/docs/main/merge-semantics.md

```diff
@@ -0,0 +1,269 @@
+or by changing the application's editing code (transactions and constraints) to circumvent it.
+
+## A Holistic View
+
+`SharedTree` allows developers to describe the data model for their application in terms of a set of elementary building blocks,
+like objects nodes, map nodes, and array nodes.
+Each of these building blocks comes with its own editing API and associated merge semantics,
+so fully understanding the merge semantics of `SharedTree` entails understanding the individual merge semantics of each of these building blocks.
+That said, those merge semantics are in large part underpinned by a set of design choices that apply to `SharedTree` as a whole.
+Understanding those design choices and their ramifications is the most crucial part of understanding `SharedTree`'s merge semantics,
+and often alleviates the need to remember the details of any building blocks's specific merge semantics.
+
+### Movement Is Not Copy
+
+`SharedTree` allows subtrees to be moved from one location to another,
+and such movement does not adversely affect concurrent edits that target the moved subtree.
+This is different from inserting a new copy of the moved subtree at the destination
+(and deleting the original at the source).
+
+Consider the following scenario:
+Alice moves a sticky note from one list to another,
+while Bob concurrently edits the text of the note.
+If the move were just a copy, then if Alice's edit were to be sequenced first,
+Bob's edit would not apply to the copy at the destination.
+By contrast, `SharedTree`'s move semantics ensure that Bob's edit will apply no matter the sequencing order.
+
+### Minimal Preconditions
+
+In the context of merge semantics,
+preconditions can be used to prevent an edit/transaction from being applied
+if concurrent edits that were sequenced earlier have impacted the document in a way that violates some specific requirements.
+
+By default, that is, in the absence of [constraints](#constraints),
+the only precondition that must be met in order for an edit to apply,
+is that the edit must not violate the document's schema.
+
+For example,
+when moving a node to a specific destination only requires that the node's type be allowed by the schema of the destination.
+
+One could imagine a host of potential additional preconditions for such an edit:
+
+-   The node being moved must not have been concurrently moved
+-   The node being moved must not have been concurrently removed
+-   Ancestors of the node being moved must not have been concurrently removed
+-   The parent and ancestors of destination must not have been concurrently removed
+
+`SharedTree`'s merge semantics are such that, by default, only the schema matters.
+The benefits of not violating the schema are obvious,
+but the benefits of not having more preconditions are more subtle.
+Indeed, while no one should wish for their edit to violate the schema,
+one may wish for their edit not to be carried out if some other precondition is not met.
```

Comment:
"While" gives an expectation that the two clauses are in some kind of opposition, but they don't seem to be.

Reply:
I rephrased the section